backoff/jitter.go

```diff
@@ -0,0 +1,14 @@
+package backoff
+
+import "time"
+
+type Jitter func(randomGen RandomGenerator, n int64) time.Duration
```

Comment:
I would rename `n` to `factor` since imo it's more common occurrence in other jitter implementations

Reply:
Renamed and simplified backoff as discussed